### PR TITLE
model: separate out TargetSpec and TargetStatus

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc, @jazzdan,

Please review the following commits I made in branch nicks/nesting:

0fee9b66402849fe1586e6863606ab18e85f3e35 (2019-01-10 14:51:43 -0500)
model: separate out TargetSpec and TargetStatus

### CODE REVIEW

File: internal/store/engine_state.go [hunk 1]
```diff
@@ -69,6 +71,70 @@ type EngineState struct {
...
+	_, ok := e.ManifestTargets[mn]
+	if !ok {
+		e.ManifestDefinitionOrder = append(e.ManifestDefinitionOrder, mn)
+	}
+	e.ManifestTargets[mn] = mt
+}
+
+func (e EngineState) Manifest(mn model.ManifestName) (model.Manifest, bool) {
+	m, ok := e.ManifestTargets[mn]
+	if !ok {
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/store/engine_state.go]: nope, that won't work. `m` is a `*ManifestTarget`, not a Manifest

File: internal/engine/buildcontroller.go [hunk 2]
```diff
@@ -102,8 +91,8 @@ func (c *BuildController) needsBuild(ctx context.Context, st store.RStore) (buil
 	}
 
 	c.lastActionCount = state.BuildControllerActionCount
-	ms := state.ManifestStates[mn]
-	manifest := ms.Manifest
+	ms, _ := state.ManifestState(mn)
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/engine/buildcontroller.go]: i just totally changed the function signature to return a Target

File: internal/engine/buildcontroller.go [hunk 3]
```diff
@@ -102,8 +91,8 @@ func (c *BuildController) needsBuild(ctx context.Context, st store.RStore) (buil
...
+	manifest, _ := state.Manifest(mn)
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/engine/buildcontroller.go]: moot

File: internal/store/engine_state.go [hunk 4]
```diff
@@ -90,7 +156,7 @@ type ResourceState interface {
 }
 
 type ManifestState struct {
-	Manifest model.Manifest
+	Name model.ManifestName
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/store/engine_state.go]: done

File: internal/store/manifest_target.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package store
+
+import "github.com/windmilleng/tilt/internal/model"
+
+type ManifestTarget struct {
+	Manifest model.Manifest
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/store/manifest_target.go]: i don't know! we'll see. trying to make incremental changes to minimize merge conflicts

[the remark this answers is not in the document]
Comment (nicks) [reply internal/store/manifest_target.go]: (i'm not even convinced at this point that the name `Manifest` will stick around)

File: internal/store/engine_state.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (maiamcc) [reply internal/store/engine_state.go]: oh shoot you're right! 😅

**Pull request merged**